src/Firebase/Auth/ApiClient.php

```diff
@@ -191,22 +191,25 @@ public function verifyPassword(string $email, string $password): ResponseInterfa
 
     /**
      * @param string $idToken
+     * @param string $continueUrl
      *
      * @return ResponseInterface
      */
-    public function sendEmailVerification(string $idToken): ResponseInterface
+    public function sendEmailVerification(string $idToken, string $continueUrl = null): ResponseInterface
     {
         return $this->request('getOobConfirmationCode', [
             'requestType' => 'VERIFY_EMAIL',
             'idToken' => $idToken,
+            'continueUrl' => $continueUrl,
```

Comment:
Please wrap the array in `array_filter(...)` to avoid sending an empty `continueUrl`field if none is set (the small things :D)

Reply:
If I do that, am I OK to move the array to a var and add in the URL if not null instead (or use array filter instead of the `if` check below for future changes? Otherwise I'll have to add the array_filter inside the function *and* have a custom function so it doesn't remove an ID token of '0' which will look a bit messy?

```php
$data = [
    'requestType' => 'VERIFY_EMAIL',
    'idToken' => $idToken,
];

if ($continueUrl !== null) {
    $data['continueUrl'] = $continueUrl;
}

return $this->request('getOobConfirmationCode', $data);
```

Reply:
Wrapping it with `array_filter()` really is sufficient - a value of `0` would be rejected by the Firebase Auth API anyways.

Reply:
'0' is valid for Firebase.
![image](https://user-images.githubusercontent.com/151294/42704941-a340046a-86c9-11e8-81bd-ffc1206aa7d9.png)

Reply:
And that explains my confusion - this function has a token not a UID so by this point it would never be a UID. For some reason I had it in my head this function had the user ID!